### ISSUE TITLE
Replace Deprecated Test Case Assertion Methods in twisted.names twisted.news and twisted.python

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#6222]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)| @Julian reported|
|-|-|
|Trac ID|trac#6222|
|Type|enhancement|
|Created|2012-12-26 03:35:38Z|
|Branch|https://github.com/twisted/twisted/tree/remove-deprecated-test-methods-names-6222|

See [#5771](https://github.com/twisted/twisted/issues/5771)

Attachments:

* [names-news-python-deprecate.patch](https://raw.githubusercontent.com/twisted/twistedmatrix.com-trac-attachments/trunk/ticket/e6b/e6b5d65892d7e32d18f4e380865a861d34543698/ba0b0a287df8ef62f52d4b0d70f29aa7e91546fa.patch) (7790 bytes) - added by Julian on 2012-12-26 03:35:46Z - 

<details><summary>Searchable metadata</summary>

```
trac-id__6222 6222
type__enhancement enhancement
reporter__Julian Julian
priority__normal normal
milestone__ 
branch__branches_remove_deprecated_test_methods_names_6222 branches/remove-deprecated-test-methods-names-6222
branch_author__julian julian
status__closed closed
resolution__fixed fixed
component__core core
keywords__ 
time__1356492938000000 1356492938000000
changetime__1388589659000000 1388589659000000
version__None None
owner__rwall rwall

```
</details>

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)<a name="note_1"></a>|@tomprince commented|
|-|-|

There are a couple of methods to be deprecated that haven't been changed. At least ```assertIdentical``` and ```assertNotIdentical```.

**Comment by twisted-trac** [comment]
|[<img alt="tomprince's avatar" src="https://avatars.githubusercontent.com/u/283816?s=50" width="50" height="50">](https://github.com/tomprince)|@tomprince set owner to @Julian|
|-|-|

**Comment by twisted-trac** [comment]
|<img alt="julian's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_3"></a>|julian commented|
|-|-|

(In [39018]) Branching to remove-deprecated-test-methods-names-6222.

**Comment by twisted-trac** [comment]
|[<img alt="Julian's avatar" src="https://avatars.githubusercontent.com/u/329822?s=50" width="50" height="50">](https://github.com/Julian)<a name="note_4"></a>|@Julian removed owner|
|-|-|

[Build results](http://buildbot.twistedmatrix.com/boxes-supported?branch=/branches/remove-deprecated-test-methods-names-6222)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_5"></a>|@exarkun set owner to @wallrj|
|-|-|

Thanks.

I guess this is an example of a change that would be easier if we actually did the `PendingDeprecationWarning` step in CompatibilityPolicy.  If we did, then I could turn on that warning type and run the test suite and observe that no warnings are emitted.

Oh well.  This change looks great.  If by some chance some method was missed then we can always deal with it later on.  Please merge!

**Comment by twisted-trac** [comment]
|<img alt="julian's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50"><a name="note_6"></a>|julian set status to `closed`|
|-|-|

(In [41086]) Merge remove-deprecated-test-methods-names-6222: Remove to-be-deprecated test methods from twisted.names

Author: Julian
Reviewers: tom.prince, exarkun
Fixes: [#6222](https://github.com/twisted/twisted/issues/6222)

Also removes the few from twisted.python. See [#5771](https://github.com/twisted/twisted/issues/5771) for more info.
